### DISCUSSION TITLE
Q: how to distinguish a literal from a variable

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#260] [Q&A] [answered]
Classic Rexx has the (unwieldy? Early documentation was severely backpedalled later from using unquoted strings (what I call literals) and an option strictvalue was introduced, as is a 'signal on novalue') literal strings to consider. Now looking at the Datatype() bif and the consequences of Unicode and cREXX level B, I wonder if we need a call to see if a string fed to it does exist as a variable name or is a literal. In case of a variable name, it can be a string, int or float, and we can inspect that with gettp - we should use the right value for the rest of datatype (which really is some shorthand for various forms of 'verify()').

**Comment by rvjansen** [comment, marked as answer]
Answering this: we need the SYMBOL() BIF. I opened https://github.com/adesutherland/CREXX/issues/261

**Reply by adesutherland** [reply]
I will need to think about this - but my first feeling is that this is a special case of introspection. More generally we will need to be able to see what an object is (what type it is and what interfaces it supports)
